File: findmeclinic/FindMeClinic/src/app/bookappointment.ts
import { Patient } from './Patient';
import { Doctor } from './doctor';

export class BookAppointment {
    
    id:String;
    patient:Patient;
    doctor:Doctor;
    appointmentDate:Date;
    slot:String;
    appointmentId:number;
    key:String;
    appointmentTime:string;
}

File: bookappointment-service/src/main/java/com/stackroute/domain/BookAppointment.java
package com.stackroute.domain;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.Transient;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.Date;

@Document
@Data
@NoArgsConstructor
@AllArgsConstructor
public class BookAppointment
{

    @Id
    private String id;
    private Doctor doctor;
    private Patient patient;
    private Date appointmentDate;
    private Integer appointmentId;
    private String appointmentTime;
    private String slot;
    @Transient
    private String key;

}
